File: c++/bubbleSort.cpp
//sort a god damn array

#include <bits/stdc++.h>


using namespace std;


void b_sort(int arr[], int length) {

    bool swapped;

    for (int i = 0; i < length; i++) {
        swapped = false;
        for (int j = 0; j < length - 1; j++) {

            if (arr[j] > arr [j+1]) {
                swap(arr[j], arr[j+1]);
                swapped = true;
            }
        }
        if (!swapped)
            break;
    }
}

void print_array(int arr[], int size) {

    int i;
    for (i = 0; i < size; i++)
        cout << " " << arr[i];

}

int main() {
    int unsorted_array[10] = {5,4,3,2,9,8,7,6,1,0};
    int n = sizeof(unsorted_array) / sizeof(unsorted_array[0]);
    b_sort(unsorted_array,n);
    print_array(unsorted_array, 10);

    return 0;

}

File: pyt/bubbleSort.py
def b_sort(arr, len):
    for i in range(len - 1):
        swapped = False
        for j in range(0, len - 1):

            if arr[j] > arr[j + 1]:
                swapped = True
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

        if not swapped:
            return


if __name__ == '__main__':
    arr = [3, 213321, 3312, 43242, 5, 323, 435237627, 2121, 1223, 1]
    n = len(arr)
    b_sort(arr, n)
    print(arr)
